APMrover2/commands_logic.cpp

```diff
@@ -122,6 +122,25 @@ void Rover::exit_mission()
     }
 }
 
+// verify_command_callback - callback function called from ap-mission at 10hz or higher when a command is being run
+//      we double check that the flight mode is AUTO to avoid the possibility of ap-mission triggering actions while we're not in AUTO mode
+bool Rover::verify_command_callback(const AP_Mission::Mission_Command& cmd)
+{
+    if (control_mode == AUTO) {
+        bool cmd_complete = verify_command(cmd);
+
+        // send message to GCS
+        if (cmd_complete) {
+            gcs_send_message(MSG_MISSION_ITEM_REACHED);
+        }
+
+        return cmd_complete;
+    }
+
+    // exit immediately if not in AUTO mode
```

Comment:
agreed, all should return false. However, as Randy as noted this callback can't even get called when not in auto so the whole control_mode == AUTO check, and the else return false are useless code. If we want to remove any of that lets do it in a separate PR